spec/amber/router/params_spec.cr

```diff
@@ -0,0 +1,84 @@
+require "../../../spec_helper"
+
+module Amber::Router
+  describe Parser do
+    headers = HTTP::Headers.new
+
+    context "when parsing query string" do
+      it "returns query params" do
+        request = HTTP::Request.new("GET", "/?test=test")
+        request.params["test"].should eq "test"
+      end
+
+      it "returns array of values for same param key" do
+        request = HTTP::Request.new("GET", "/?test=test&test=test2")
+        request.params.fetch_all("test").size.should eq 2
+        request.params.fetch_all("test").should eq %w(test test2)
+      end
+    end
+
+    context "when parsing request body" do
+      context "when content-type is form url urlencoded" do
+        it "returns params" do
+          headers["Content-Type"] = "application/x-www-form-urlencoded"
+          body = "name=John Doe"
+          request = HTTP::Request.new("POST", "/", headers, body)
+          request.params["name"].should eq "John Doe"
+        end
+
+        it "parses body params with charset" do
+          headers["Content-Type"] = "application/x-www-form-urlencoded; charset=UTF-8"
+          body = "\x68\x65\x6C\x6C\x6F\x3D\x77\x6F\x72\x6C\x64"
```

Comment:
Use `Bytes[104, 101, 108, 108, 111, 61, 119, 111, 114, 108, 100]` instead of broken string. @RX14 and <Oprypin> are pretty adamant that non utf8 strings should not be used.